core/staking_verifier.go

```diff
@@ -212,15 +217,21 @@ func VerifyAndDelegateFromMsg(
 		return nil, nil, nil, errNegativeAmount
 	}
 	if msg.Amount.Cmp(minimumDelegation) < 0 {
-		return nil, nil, nil, errDelegationTooSmall
+		if chainConfig.IsMinDelegation100(epoch) {
+			if msg.Amount.Cmp(minimumDelegationV2) < 0 {
+				return nil, nil, nil, errDelegationTooSmallV2
+			}
+		} else {
+			return nil, nil, nil, errDelegationTooSmall
+		}
 	}
 
 	updatedValidatorWrappers := []*staking.ValidatorWrapper{}
 	delegateBalance := big.NewInt(0).Set(msg.Amount)
 	fromLockedTokens := map[common.Address]*big.Int{}
 
 	var delegateeWrapper *staking.ValidatorWrapper
-	if redelegation {
+	if chainConfig.IsRedelegation(epoch) {
```

Comment:
I noticed that there are now functional adjustments about chains, are they all based on epoch Instead of a specific height value? This feels like it would be more convenient, an epoch I remember is 24 hours?

Reply:
all of our hard forks are based on epochs instead of blocks.

Reply:
sounds great